src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs

```diff
@@ -582,5 +655,227 @@ protected static int GetParameterIndex<TNode>(SeparatedSyntaxList<TNode> paramet
 
             return parameters.Count - 1;
         }
+
+        protected (IEnumerable<T> parameters, IEnumerable<SyntaxToken> separators) UpdateDeclarationBase<T>(
+            SeparatedSyntaxList<T> list,
+            SignatureChange updatedSignature,
+            Func<AddedParameter, T> createNewParameterMethod) where T : SyntaxNode
+        {
+            var originalParameters = updatedSignature.OriginalConfiguration.ToListOfParameters();
+            var reorderedParameters = updatedSignature.UpdatedConfiguration.ToListOfParameters();
+
+            int numAddedParameters = 0;
+
+            var newParameters = new List<T>();
+            for (var index = 0; index < reorderedParameters.Count; index++)
+            {
+                var newParam = reorderedParameters[index];
+                if (newParam is ExistingParameter existingParameter)
+                {
+                    var pos = originalParameters.IndexOf(p => p is ExistingParameter ep && ep.Symbol == existingParameter.Symbol);
+                    if (pos >= 0)
+                    {
+                        var param = list[pos];
+
+                        // copy whitespace trivia from original position
+                        param = TransferLeadingWhitespaceTrivia(param, list[index - numAddedParameters]);
```

Comment:
Is there special handling needed here for extension methods? Because that won't be in the SeparatedSyntaxList?

Reply:
> Because that won't be in the SeparatedSyntaxList?

Why wouldn't it be? This is for declarations so it can't be omitted or anything...